pallets/identity/src/lib.rs

```diff
@@ -423,10 +423,22 @@ decl_module! {
                 |batch_claim_item| <DidRecords>::contains_key(batch_claim_item.target)),
                 Error::<T>::DidMustAlreadyExist);
 
+            let cdd_providers = T::CddServiceProviders::get_members();
+            let cdd_count: usize = claims
+                .iter()
+                .filter(|batch_claim_item| match batch_claim_item.claim {
+                    Claim::CustomerDueDiligence => true,
+                    _ => false,
+                })
+                .count();
+            if cdd_count > 0 { 
+                ensure!(cdd_providers.contains(&issuer), Error::<T>::UnAuthorizedCddProvider);
+            }
+
             T::ProtocolFee::charge_fee_batch(
                 &Signatory::AccountKey(sender_key),
                 ProtocolOp::IdentityAddClaim,
-                claims.len()
+                claims.len() - cdd_count
```

Comment:
We might want to still charge protocol fees for all claims, including CDD claims. Cc @adamdossa.

Reply:
Currently we bypass the fee when adding CDD claims in other places, so I think we should be consistent here. This logic is a bit TBD and may be revisited.